src/plugins/fs.c

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (C) 2016  Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Vratislav Podzimek <vpodzime@redhat.com>
+ */
+
+#include <exec.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <blkid.h>
+
+#include "fs.h"
+
+/**
+ * SECTION: fs
+ * @short_description: plugin for operations with file systems
+ * @title: FS
+ * @include: fs.h
+ *
+ * A plugin for operations with file systems
+ */
+
+/**
+ * bd_fs_error_quark: (skip)
+ */
+GQuark bd_fs_error_quark (void)
+{
+    return g_quark_from_static_string ("g-bd-fs-error-quark");
+}
+
+/**
+ * bd_fs_ext4_info_copy: (skip)
+ *
+ * Creates a new copy of @data.
+ */
+BDFSExt4Info* bd_fs_ext4_info_copy (BDFSExt4Info *data) {
+    BDFSExt4Info *ret = g_new0 (BDFSExt4Info, 1);
+
+    ret->label = g_strdup (data->label);
+    ret->uuid = g_strdup (data->uuid);
+    ret->state = g_strdup (data->state);
+    ret->block_size = data->block_size;
+    ret->block_count = data->block_count;
+    ret->free_blocks = data->free_blocks;
+    ret->mount_count = data->mount_count;
```

Comment:
Why include mount_count if not including maximal mount count or whatever it's called?